File: cpp/0046全排列.cpp
#include<vector>
#include<algorithm>

using namespace std;
vector<int> temp;
vector<vector<int>> res;

void BackTracking(const vector<int> &nums)
{
    if(temp.size()==nums.size())
    {
        res.push_back(temp);
        return;
    }

    for(int i :nums)
    {
        if(find(temp.begin(),temp.end(),i)!=temp.end())
            continue;
        temp.push_back(i);
        BackTracking(nums);
        temp.pop_back();
    }
}

vector<vector<int>> permute(vector<int> &nums)
{
    BackTracking(nums);
    return res;
}

File: python/main/0046全排列.py

def permute(nums:list) -> list:
    res=[]
    temp=[]

    def _backTracking():
        nonlocal nums

        if(len(temp)==len(nums)):
            res.append(temp.copy())
            return
            
        for i in nums:
            if(i not in temp):
                temp.append(i)
                _backTracking()
                temp.pop()
    
    _backTracking()
    return res


print(permute([1,2,3]))